src/libraries/System.Text.Json/src/System/Text/Json/Serialization/JsonSerializer.Read.HandleMetadata.cs

```diff
@@ -12,17 +12,22 @@ public static partial class JsonSerializer
         internal static readonly byte[] s_idPropertyName
             = new byte[] { (byte)'$', (byte)'i', (byte)'d' };
 
-        internal static ReadOnlySpan<byte> s_refPropertyName
-            => new byte[] { (byte)'$', (byte)'r', (byte)'e', (byte)'f' };
+        internal static readonly byte[] s_refPropertyName
+            = new byte[] { (byte)'$', (byte)'r', (byte)'e', (byte)'f' };
+
+        internal static readonly byte[] s_valuesPropertyName
+            = new byte[] { (byte)'$', (byte)'v', (byte)'a', (byte)'l', (byte)'u', (byte)'e', (byte)'s' };
 
         /// <summary>
         /// Returns true if successful, false is the reader ran out of buffer.
-        /// Sets state.Current.ReturnValue to the $ref target for MetadataRefProperty cases.
+        /// Sets state.Current.ReturnValue to the reference target for $ref cases;
+        /// Sets state.Current.ReturnValue to a new instance for $id cases.
         /// </summary>
-        internal static bool ResolveMetadata(
-            JsonConverter converter,
+        internal static bool ResolveMetadataForJsonObject<T>(
+            JsonResumableConverter<T> converter,
```

Comment:
Does this need to be generic? Can we put the virtual method on `JsonConverter`? Same below.

Reply:
Done, thanks for the suggestion.